src/components/PageHeader.tsx

```diff
@@ -168,17 +177,41 @@ export const DashboardPageHeader = () => {
         console.log('Selected: ', activeItem);
         setActiveItem(result.itemId);
     };
+    const [
+        getAuthZ,
+        { loading: _authzLoading, error: _authzError, data: authzData },
+    ] = useLazyQuery<AuthZMapping>(AuthZMappingQuery, {
+        errorPolicy: 'all',
+        /* need to re-fetch each time when user press save/back button */
+        fetchPolicy: 'network-only',
+        notifyOnNetworkStatusChange: true,
+    });
+    useEffect(() => {
+        getAuthZ();
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
     /** Horizontal menu */
-    const menuElements = menuRoutes.map((item) => (
-        <NavItem
-            itemId={item.key}
-            key={item.key}
-            to={`#${item.to}`}
-            isActive={item.to === location.pathname}
-        >
-            {item.title}
-        </NavItem>
-    ));
+    const menuElements = _(menuRoutes)
+        .map((item) => {
```

Comment:
I do not know what to reply.
Is this NACK for merge from your side because of this?
Code can be written in many different ways.
Improvements/contributions are welcome.

Reply:
OK, I'll be more explicit: Please rewrite this so that it's `.filter()` then `.map()`. It's much easier to read “pick items for which the user has authorisation and make them into menu buttons” than it is to read “make the unauthorised items null and the others into menu buttons and then get rid of all the nulls”.

What do you mean by "Improvements/contributions are welcome"? I'm trying to be helpful in this review so that the code is readable, easy to maintain and extend and so that it prevents unnecessary bugs.

Reply:
I do not want to rewrite, since original code is valid.
Contributions are welcome.
Be picky doesn't help.
Please prove that there is: "unnecessary bug". You provoke me again here.

Reply:
Just to tell you that I do not have a clear opinion on that since I don't have lot of knowledge here. But it would be nice if you both could write down what are the pros and cons of filter -> map vs. map -> filter. If it is just this type of situation where we are trying to find out if we should do `for line in f:` or `for line in f.readlines():`, then I do not care which one is merged and it is not a blocker for me. But as I said, I cannot really decide on that.

Reply:
We agreed with @mgrabovsky to keep this code. Mark as resolved.